acceptance/tests/ticket_1238_hostname_fqdn.rb

```diff
@@ -0,0 +1,22 @@
+test_name 'ticket 1238 facter should respect hostname if fqdn'
+
+confine :except, :platform => 'windows'
+
+fqdn = 'foo.bar.example.org'
+
+agents.each do |host|
```

Comment:
i prefer the iterator var to be agent in these cases, so it's easier for others to read the code; eg for the cases when one uses a hosts loop.  but maybe only important if you make other changes to this PR